lib/request-handler.js

```diff
@@ -16,20 +16,22 @@ var retryOnNextHostDecision = {
 
 /**
  * Handles a request to Cassandra, dealing with host fail-over and retries on error
- * @param {?Client} client Client instance used to retrieve and set the keyspace. It can be null.
+ * @param {Client} client Client instance used to retrieve and set the keyspace.
  * @param {Object} options
  * @constructor
  */
 function RequestHandler(client, options) {
   this.client = client;
-  this.loadBalancingPolicy = options.policies.loadBalancing;
-  this.retryPolicy = options.policies.retry;
+  this.profileManager = client.profileManager;
+  var profile = client.profileManager.getDefault();
+  this.loadBalancingPolicy = profile.loadBalancing;
```

Comment:
`RequestHandler` has 3 public methods: `#send()`, `#prepareMultiple()` (preparing for batches) and `setKeyspace()` (not so easy to see, I've created [NODEJS-266][NODEJS-266] to use proper notation for private methods).

For all cases, the load-balancing policy (and retry policy) should be the one defined in the query options (inheriting from profiles) and be the same for as long `RequestHandler` lives (`final`/`readonly` in java or C#). Your doubts are valid, its not clear in the code: I'll change it to instance fields initiated on the `RequestHandler`.

[NODEJS-266]: https://datastax-oss.atlassian.net/browse/NODEJS-266